src/NamedTupleUtilities.jl

```diff
@@ -51,13 +51,44 @@ julia> range((a=1,b=2,c=3),Val(:a),Val(:b))
             push!(names, n)
         end
         if n == cn
+            # rangeStarted = false
+            break
+        end
+    end
+    types = Tuple{(fieldtype(a, n) for n in names)...}
+    vals = Expr[:(getfield(a, $(QuoteNode(n)))) for n in names]
+    return :( NamedTuple{$(names...,),$types}(($(vals...),)) )
+end
+
+"""
+    range(a::NamedTuple, b::Val{n}, c::Val{n})
+Return a NamedTuple which retains `b`th to `c`th fields in `a`. 
+If `b` is greater than or equal to `c`, then it will return the empty NamedTuple.
+```jldoctest
+julia> QueryOperators.NamedTupleUtilities.range((a=1,b=2,c=3),1,2)
+(a = 1, b = 2)
+```
+"""
+@generated function range(a::NamedTuple{an}, b::Int, c::Int) where {an}
+    rangeStarted = false
+    names = Symbol[]
+    count = 1
+    for n in an
+        if count == b
```

Comment:
With the current version, I believe ``b`` will have the value ``Int``, and comparing that with ``1`` will always be ``false``. The issue is that in a generated function, you *never* have access to the values passed, you always just get the types of the arguments. We can get around that by using the ``Val{b}`` method signature I indicated above.

Also, we should write a unit test, I believe that should have caught the issue here?